### PR TITLE
PostDiffViewer: Move component to /blocks/

### PR DESCRIPTION
**Pull request by spen**
*Note*: This PR is currently point to #20822 as it's base - that should be landed prior to this PR.

### Summary
This PR is the second in an effort towards decoupling some components that have been built during the development of the post revisions work, making them reusable in other areas of the project, easier to unit test and easier to document.

Specifically - here we're moving `EditorDiffViewer` in to a more general-use `PostDiffViewer` block.

screen shot 2017-12-14 at 20 46 33

### Testing
There should be no deviations from the original functionality in post revisions.
To test this:
- Go to a the post editor
- Make some revisions if the post doesn't already have a few
- Click on the 'history' button in the editors header
- You should see removals highlighted in a faint red, additions in a faint blue exactly as before
- Also pay attention to the line splitting, the title shouldn't break changes on to new lines but the changes in the content should.

Also, take a look at the DevDocs entry - does this look reasonable?
Go to https://calypso.live/devdocs and search PostDiffViewer under 'blocks'.

Note: I'm open to renaming this block, it's a tough one to name given it applies to pages and CPT's as well as standard posts.

### CODE REVIEW

File: client/blocks/post-diff-viewer/docs/example.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			{ op: 'del', value: '.' },
+			{ op: 'add', value: '!' },
+		],
+		content: [
+			{ op: 'copy', value: 'Today the price of Bitcoin ' },
+			{ op: 'del', value: 'climbed' },
+			{ op: 'add', value: 'rocketed' },
+			{ op: 'copy', value: ' to over ' },
+			{ op: 'del', value: '$15000' },
+			{ op: 'add', value: '$16000' },
```
Comment (jblz) [inline client/blocks/post-diff-viewer/docs/example.jsx]: And to think, it was over 17 just yesterday!

Comment (marekhrabe) [reply client/blocks/post-diff-viewer/docs/example.jsx]: Let's connect this example to a realtime API! :D

Comment (spen) [reply client/blocks/post-diff-viewer/docs/example.jsx]: Saving the `<CorrectMyBitcoin />` component for another PR 😄

**Comment by spen** [comment]
With the recent merges there's now a ton more to be done to generalise this component. Given that it's currently only used in one place I'll close this out until the time comes that we need it elsewhere :)

**Pull request closed without merging**